### PR TITLE
aws_ec2 inventory plugin: added ability for using Jinja on AWS credentials

### PR DESCRIPTION
**Pull request by goneri**
##### SUMMARY

https://github.com/ansible-collections/amazon.aws/pull/57 with a changelog and a couple of fixes.

closes: #57

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

File: changelogs/fragments/57_inventory_accept_Jinja2_template.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+- aws_ec2 inventory plugin accepts Jinja2 template in the authentication fields
```
Comment (Akasurde) [inline changelogs/fragments/57_inventory_accept_Jinja2_template.yaml:3]:
```suggestion
- ``aws_ec2`` inventory plugin accepts Jinja2 template in the authentication fields (https://github.com/ansible-collections/amazon.aws/pull/227).
```

File: plugins/inventory/aws_ec2.py [hunk 2]
```diff
@@ -577,16 +581,25 @@ def _add_hosts(self, hosts, group, hostnames):
...
+        credentials = {}
+
+        for credential_type in ['aws_profile', 'aws_access_key', 'aws_secret_key', 'aws_security_token', 'iam_role_arn']:
+            # This step won't be necessary with https://github.com/ansible/ansible/pull/58288
+            if self.templar.is_template(self.get_option(credential_type)):
+                credentials[credential_type] = self.templar.template(variable=self.get_option(credential_type), disable_lookups=False)
+            else:
+                credentials[credential_type] = self.get_option(credential_type)
+
+        self.boto_profile = credentials['aws_profile']
```
Comment (Akasurde) [inline plugins/inventory/aws_ec2.py:598]:
```suggestion
        self.boto_profile = credentials.get('aws_profile')
```

Comment (Akasurde) [reply plugins/inventory/aws_ec2.py:598]: https://docs.quantifiedcode.com/python-anti-patterns/correctness/not_using_get_to_return_a_default_value_from_a_dictionary.html?highlight=get#use-dict-get-key-default-to-assign-default-values

Comment (goneri) [reply plugins/inventory/aws_ec2.py:598]: I think the original version is fine. `credentials['aws_profile']` should always be defined. If we use `.get()` here we may actually miss some problem with the block above.

File: changelogs/fragments/57_inventory_accept_Jinja2_template.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+- "``aws_ec2`` inventory plugin accepts Jinja2 template in the authentication fields (https://github.com/ansible-collections/amazon.aws/pull/227)."
```
Comment (tremble) [inline changelogs/fragments/57_inventory_accept_Jinja2_template.yaml:3]:
```suggestion
- "aws_ec2 inventory plugin - Added support for using Jinja2 templates in the authentication fields (https://github.com/ansible-collections/amazon.aws/pull/227)."
```

**Review by tremble — CHANGES_REQUESTED** [review]
I'd suggest a minor tweak to the changelog fragment, otherwise looks fine.  Not confident with the inventory plugin architecture and templating, so I'd prefer @jillr someone from the cloud team also reviews (@jillr)

**Comment by tremble** [comment]
Looking back at the previous PR, my other review comment still stands - We need to add some integration tests

File: plugins/inventory/aws_ec2.py [hunk 4]
```diff
@@ -577,16 +581,25 @@ def _add_hosts(self, hosts, group, hostnames):
             # Create groups based on variable values and add the corresponding hosts to it
             self._add_host_to_keyed_groups(self.get_option('keyed_groups'), host, hostname, strict=strict)
 
-    def _set_credentials(self):
+    def _set_credentials(self, loader):
```
[the remark this answers is not in the document]
Comment (goneri) [reply plugins/inventory/aws_ec2.py]: Indeed. Let's fix that.

**Comment by tremble** [comment]
Since this PR's suffered from bit-rot I'm going to close it out.

I've revived #57 and added a simple integration test

**Pull request closed without merging**